lib/saluki-app/src/api.rs

```diff
@@ -64,10 +112,18 @@ impl APIBuilder {
 
         // We have to convert this Tower-based `Service` to a Hyper-based `Service` to use it with `HttpServer`, since
         // the two traits are different from a semver perspective.
-        let service = TowerToHyperService::new(self.router);
+        let http_service = self.router;
+
+        let multiplexed_service = TowerToHyperService::new(MultiplexService::new(
+            http_service,
+            self.grpc_router.unwrap().into_router(),
```

Comment:
We shouldn't unwrap here. Maybe just `.unwrap_or_else(|| Server::builder())`?

Reply:
Hmm, I believe `Server::builder()` needs a `.add_service(svc)` call on it to actually make it return a `Router` which we would then use with the multiplexed service. Otherwise it just returns a `Server`

Reply:
Hmm, doh, that's true.

Looking at the docs, I think we might want to change `grpc_router` to be [`RoutesBuilder`](https://docs.rs/tonic/latest/tonic/service/struct.RoutesBuilder.html): this would let us initialize the field with `RoutesBuilder::default()`, add multiple gRPC services via `RoutesBuilder::add_service`, and then in `serve`, we could consume it and get a `Routes` by calling `RoutesBuilder::routes`, which we would then convert into the proper router type by calling `Routes::into_axum_router`.